Upgrade fixie to 3.0.0-beta.1

chan18 opened the pull request:
upgradeing fixie to 3.0.0-beta.1

chan18 commented:
Hi, @plioi is this something related to the Fixie to 3.0.0-beta.1, Do you have any suggestions?

plioi commented:
I'm not sure specifically, but I think that I'm not seeing the code that produced this message. Navigating from your link, I see Directory.build.props still says Fixie 2.x. That's what's being referred to in ContactList.Tests.csproj. But if the error message is talking about Fixie.Console 3.x, then I'm not convinced that I'm looking at the code that produced this error.

Compare the csproj for the tests project on your environment against the one here for instance: https://github.com/fixie/fixie.benchmark/blob/develop/src/Fixie.Tests/Fixie.Tests.csproj Note how there's no longer a use for the old "DotNetCliToolReference" tag, instead just regular PackageReference. Also note that Fixie.Console does NOT get installed into the projects any more, that's a "dotnet local tool" now.

Basically, compare what you have here against what you already had working in the other repos, and see if differences stand out to you.

chan18 commented:
hi, @plioi after adding the .config file and removed the local project reference for the fixie, it did build but test cases are failing and it throws a build error at the end, below is the log. I have added a complete log.


```

.\build.cmd

.NET SDK
3.1.407

Clean

Restore (Database Migration)
  Determining projects to restore...
  Restored H:\open-source\fixie.demo\src\ContactList.DatabaseMigration\ContactList.DatabaseMigration.csproj (in 175 ms).

Restore (Solution)
  Determining projects to restore...
  Restored H:\open-source\fixie.demo\src\ContactList.DatabaseMigration\ContactList.DatabaseMigration.csproj (in 192 ms).
  2 of 3 projects are up-to-date for restore.

Build
  ContactList.DatabaseMigration -> H:\open-source\fixie.demo\src\ContactList.DatabaseMigration\bin\Release\netcoreapp3.1\ContactList.DatabaseMigration.dll
  ContactList -> H:\open-source\fixie.demo\src\ContactList\bin\Release\netcoreapp3.1\ContactList.dll
  ContactList -> H:\open-source\fixie.demo\src\ContactList\bin\Release\netcoreapp3.1\ContactList.Views.dll
TestingConvention.cs(21,34): warning CS8602: Dereference of a possibly null reference. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
TestingConvention.cs(25,38): warning CS8602: Dereference of a possibly null reference. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
TestingConvention.cs(25,38): warning CS8600: Converting null literal or possible null value to non-nullable type. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
TestingConvention.cs(26,65): warning CS8602: Dereference of a possibly null reference. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
TestingConvention.cs(26,46): warning CS8604: Possible null reference argument for parameter 'method' in 'object MethodInfoExtensions.Execute(MethodInfo method, object instance, params object[] parameters)'. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(18,26): warning CS8600: Converting null literal or possible null value to non-nullable type. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(22,47): warning CS8604: Possible null reference argument for parameter 'source' in 'ExceptionDispatchInfo ExceptionDispatchInfo.Capture(Exception source)'. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(27,24): warning CS8603: Possible null reference return. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(38,24): warning CS8602: Dereference of a possibly null reference. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(38,24): warning CS8603: Possible null reference return. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(41,20): warning CS8603: Possible null reference return. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(52,20): warning CS8625: Cannot convert null literal to non-nullable reference type. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
  ContactList.Tests -> H:\open-source\fixie.demo\src\ContactList.Tests\bin\Release\netcoreapp3.1\ContactList.Tests.dll

Build succeeded.

TestingConvention.cs(21,34): warning CS8602: Dereference of a possibly null reference. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
TestingConvention.cs(25,38): warning CS8602: Dereference of a possibly null reference. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
TestingConvention.cs(25,38): warning CS8600: Converting null literal or possible null value to non-nullable type. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
TestingConvention.cs(26,65): warning CS8602: Dereference of a possibly null reference. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
TestingConvention.cs(26,46): warning CS8604: Possible null reference argument for parameter 'method' in 'object MethodInfoExtensions.Execute(MethodInfo method, object instance, params object[] parameters)'. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(18,26): warning CS8600: Converting null literal or possible null value to non-nullable type. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(22,47): warning CS8604: Possible null reference argument for parameter 'source' in 'ExceptionDispatchInfo ExceptionDispatchInfo.Capture(Exception source)'. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(27,24): warning CS8603: Possible null reference return. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(38,24): warning CS8602: Dereference of a possibly null reference. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(38,24): warning CS8603: Possible null reference return. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(41,20): warning CS8603: Possible null reference return. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
MethodInfoExtensions.cs(52,20): warning CS8625: Cannot convert null literal to non-nullable reference type. [H:\open-source\fixie.demo\src\ContactList.Tests\ContactList.Tests.csproj]
    12 Warning(s)
    0 Error(s)

Time Elapsed 00:00:02.33

Update DEV/TEST Databases
Executing RoundhousE for environment: DEV
Running RoundhousE v0.9.1.0 against DESKTOP-CHANDRA\TEW_SQLEXPRESS - ContactList-DEV.
Looking in Scripts for scripts to run.
==================================================
Setup, Backup, Create/Restore/Drop
==================================================
Creating ContactList-DEV database on DESKTOP-CHANDRA\TEW_SQLEXPRESS server if it doesn't exist.
==================================================
RoundhousE Structure
==================================================
 Running database type specific tasks.
 Creating RoundhousE schema if it doesn't exist.
 Creating [Version] table if it doesn't exist.
 Creating [ScriptsRun] table if it doesn't exist.
 Creating [ScriptsRunErrors] table if it doesn't exist.
==================================================
Versioning
==================================================
 Attempting to resolve assembly file version from H:\open-source\fixie.demo\src\ContactList.DatabaseMigration\bin\Release\netcoreapp3.1\ContactList.DatabaseMigration.dll.
 Found version 1.0.0.0 from H:\open-source\fixie.demo\src\ContactList.DatabaseMigration\bin\Release\netcoreapp3.1\ContactList.DatabaseMigration.dll.
 Migrating ContactList-DEV from version 1.0.0.0 to 1.0.0.0.
 Versioning ContactList-DEV database with version 1.0.0.0 based on .
==================================================
Migration Scripts
==================================================
--------------------------------------------------
Looking for AlterDatabase scripts in "Scripts\alterDatabase".
--------------------------------------------------
--------------------------------------------------
Looking for Run Before Update scripts in "Scripts\runBeforeUp".
--------------------------------------------------
--------------------------------------------------
Looking for Update scripts in "Scripts\up". These should be one time only scripts.
--------------------------------------------------
 Skipped 2018-05-26-1255.Create_Contact.sql - One time script.
 2018-05-28-1456.Insert_Sample_Contacts.ENV.DEV.sql is an environment file. We have an environment set containing: DEV. This will run based on this check.
 Skipped 2018-05-28-1456.Insert_Sample_Contacts.ENV.DEV.sql - One time script.
--------------------------------------------------
Looking for Run First After Update scripts in "Scripts\runFirstAfterUp".
--------------------------------------------------
--------------------------------------------------
Looking for Function scripts in "Scripts\functions".
--------------------------------------------------
--------------------------------------------------
Looking for View scripts in "Scripts\views".
--------------------------------------------------
--------------------------------------------------
Looking for Stored Procedure scripts in "Scripts\sprocs".
--------------------------------------------------
--------------------------------------------------
Looking for Triggers scripts in "Scripts\triggers".
--------------------------------------------------
--------------------------------------------------
Looking for Index scripts in "Scripts\indexes".
--------------------------------------------------
--------------------------------------------------
Looking for Run after Other Anytime Scripts scripts in "Scripts\runAfterOtherAnyTimeScripts".
--------------------------------------------------
--------------------------------------------------
Looking for Permission scripts in "Scripts\permissions". These scripts will run every time
--------------------------------------------------


RoundhousE v0.9.1.0 has kicked your database (ContactList-DEV)! You are now at version 1.0.0.0. All changes and backups can be found at "H:\open-source\fixie.demo\src\ContactList.DatabaseMigration\packages\roundhouse\0.9.1\tools\output\migrations\ContactList-DEV\TEW_SQLEXPRESS\20210329_164058_6120".
Executing RoundhousE for environment: TEST
Running RoundhousE v0.9.1.0 against DESKTOP-CHANDRA\TEW_SQLEXPRESS - ContactList-TEST.
Looking in Scripts for scripts to run.
==================================================
Setup, Backup, Create/Restore/Drop
==================================================
Creating ContactList-TEST database on DESKTOP-CHANDRA\TEW_SQLEXPRESS server if it doesn't exist.
==================================================
RoundhousE Structure
==================================================
 Running database type specific tasks.
 Creating RoundhousE schema if it doesn't exist.
 Creating [Version] table if it doesn't exist.
 Creating [ScriptsRun] table if it doesn't exist.
 Creating [ScriptsRunErrors] table if it doesn't exist.
==================================================
Versioning
==================================================
 Attempting to resolve assembly file version from H:\open-source\fixie.demo\src\ContactList.DatabaseMigration\bin\Release\netcoreapp3.1\ContactList.DatabaseMigration.dll.
 Found version 1.0.0.0 from H:\open-source\fixie.demo\src\ContactList.DatabaseMigration\bin\Release\netcoreapp3.1\ContactList.DatabaseMigration.dll.
 Migrating ContactList-TEST from version 1.0.0.0 to 1.0.0.0.
 Versioning ContactList-TEST database with version 1.0.0.0 based on .
==================================================
Migration Scripts
==================================================
--------------------------------------------------
Looking for AlterDatabase scripts in "Scripts\alterDatabase".
--------------------------------------------------
--------------------------------------------------
Looking for Run Before Update scripts in "Scripts\runBeforeUp".
--------------------------------------------------
--------------------------------------------------
Looking for Update scripts in "Scripts\up". These should be one time only scripts.
--------------------------------------------------
 Skipped 2018-05-26-1255.Create_Contact.sql - One time script.
 2018-05-28-1456.Insert_Sample_Contacts.ENV.DEV.sql is an environment file. We have an environment set containing: TEST. This will NOT run based on this check.
 Skipped 2018-05-28-1456.Insert_Sample_Contacts.ENV.DEV.sql - One time script.
--------------------------------------------------
Looking for Run First After Update scripts in "Scripts\runFirstAfterUp".
--------------------------------------------------
--------------------------------------------------
Looking for Function scripts in "Scripts\functions".
--------------------------------------------------
--------------------------------------------------
Looking for View scripts in "Scripts\views".
--------------------------------------------------
--------------------------------------------------
Looking for Stored Procedure scripts in "Scripts\sprocs".
--------------------------------------------------
--------------------------------------------------
Looking for Triggers scripts in "Scripts\triggers".
--------------------------------------------------
--------------------------------------------------
Looking for Index scripts in "Scripts\indexes".
--------------------------------------------------
--------------------------------------------------
Looking for Run after Other Anytime Scripts scripts in "Scripts\runAfterOtherAnyTimeScripts".
--------------------------------------------------
--------------------------------------------------
Looking for Permission scripts in "Scripts\permissions". These scripts will run every time
--------------------------------------------------


RoundhousE v0.9.1.0 has kicked your database (ContactList-TEST)! You are now at version 1.0.0.0. All changes and backups can be found at "H:\open-source\fixie.demo\src\ContactList.DatabaseMigration\packages\roundhouse\0.9.1\tools\output\migrations\ContactList-TEST\TEW_SQLEXPRESS\20210329_164105_6562".

Test
Running ContactList.Tests

Test 'ContactList.Tests.AutoMapperConfigurationTests.ShouldBeValid' failed:

Object reference not set to an instance of an object.

System.NullReferenceException
   at ContactList.Tests.TestingConvention.ExecuteAsync(TestClass testClass) in H:\open-source\fixie.demo\src\ContactList.Tests\TestingConvention.cs:line 26
   at Fixie.TestAssembly.RunAsync() in /home/runner/work/fixie/fixie/src/Fixie/TestAssembly.cs:line 60

Test 'ContactList.Tests.Model.ContactTests.ShouldPersist' failed:

Object reference not set to an instance of an object.

System.NullReferenceException
   at ContactList.Tests.TestingConvention.ExecuteAsync(TestClass testClass) in H:\open-source\fixie.demo\src\ContactList.Tests\TestingConvention.cs:line 26
   at Fixie.TestAssembly.RunAsync() in /home/runner/work/fixie/fixie/src/Fixie/TestAssembly.cs:line 60

Test 'ContactList.Tests.Model.DatabaseTests.ShouldRollBackOnFailure' failed:

Object reference not set to an instance of an object.

System.NullReferenceException
   at ContactList.Tests.TestingConvention.ExecuteAsync(TestClass testClass) in H:\open-source\fixie.demo\src\ContactList.Tests\TestingConvention.cs:line 26
   at Fixie.TestAssembly.RunAsync() in /home/runner/work/fixie/fixie/src/Fixie/TestAssembly.cs:line 60

Test 'ContactList.Tests.Features.Contact.AddContactTests.ShouldRequireMinimumFields' failed:

Object reference not set to an instance of an object.

System.NullReferenceException
   at ContactList.Tests.TestingConvention.ExecuteAsync(TestClass testClass) in H:\open-source\fixie.demo\src\ContactList.Tests\TestingConvention.cs:line 26
   at Fixie.TestAssembly.RunAsync() in /home/runner/work/fixie/fixie/src/Fixie/TestAssembly.cs:line 60

Test 'ContactList.Tests.Features.Contact.AddContactTests.ShouldRequireValidEmailAddress' failed:

Object reference not set to an instance of an object.

System.NullReferenceException
   at ContactList.Tests.TestingConvention.ExecuteAsync(TestClass testClass) in H:\open-source\fixie.demo\src\ContactList.Tests\TestingConvention.cs:line 26
   at Fixie.TestAssembly.RunAsync() in /home/runner/work/fixie/fixie/src/Fixie/TestAssembly.cs:line 60

Test 'ContactList.Tests.Features.Contact.AddContactTests.ShouldRequireValidEmailAddress' skipped:
This test did not run.

Test 'ContactList.Tests.Features.Contact.AddContactTests.ShouldAddNewContact' failed:

Object reference not set to an instance of an object.

System.NullReferenceException
   at ContactList.Tests.TestingConvention.ExecuteAsync(TestClass testClass) in H:\open-source\fixie.demo\src\ContactList.Tests\TestingConvention.cs:line 26
   at Fixie.TestAssembly.RunAsync() in /home/runner/work/fixie/fixie/src/Fixie/TestAssembly.cs:line 60

Test 'ContactList.Tests.Features.Contact.AddContactTests.ShouldAddNewContact' skipped:
This test did not run.

Test 'ContactList.Tests.Features.Contact.ContactIndexTests.ShouldGetAllContactsSortedByName' failed:

Object reference not set to an instance of an object.

System.NullReferenceException
   at ContactList.Tests.TestingConvention.ExecuteAsync(TestClass testClass) in H:\open-source\fixie.demo\src\ContactList.Tests\TestingConvention.cs:line 26
   at Fixie.TestAssembly.RunAsync() in /home/runner/work/fixie/fixie/src/Fixie/TestAssembly.cs:line 60

Test 'ContactList.Tests.Features.Contact.DeleteContactTests.ShouldDeleteContactById' failed:

Object reference not set to an instance of an object.

System.NullReferenceException
   at ContactList.Tests.TestingConvention.ExecuteAsync(TestClass testClass) in H:\open-source\fixie.demo\src\ContactList.Tests\TestingConvention.cs:line 26
   at Fixie.TestAssembly.RunAsync() in /home/runner/work/fixie/fixie/src/Fixie/TestAssembly.cs:line 60

Test 'ContactList.Tests.Features.Contact.EditContactTests.ShouldGetCurrentContactDataById' failed:

Object reference not set to an instance of an object.

System.NullReferenceException
   at ContactList.Tests.TestingConvention.ExecuteAsync(TestClass testClass) in H:\open-source\fixie.demo\src\ContactList.Tests\TestingConvention.cs:line 26
   at Fixie.TestAssembly.RunAsync() in /home/runner/work/fixie/fixie/src/Fixie/TestAssembly.cs:line 60

Test 'ContactList.Tests.Features.Contact.EditContactTests.ShouldRequireMinimumFields' failed:

Object reference not set to an instance of an object.

System.NullReferenceException
   at ContactList.Tests.TestingConvention.ExecuteAsync(TestClass testClass) in H:\open-source\fixie.demo\src\ContactList.Tests\TestingConvention.cs:line 26
   at Fixie.TestAssembly.RunAsync() in /home/runner/work/fixie/fixie/src/Fixie/TestAssembly.cs:line 60

Test 'ContactList.Tests.Features.Contact.EditContactTests.ShouldRequireMinimumFields' skipped:
This test did not run.

Test 'ContactList.Tests.Features.Contact.EditContactTests.ShouldRequireValidEmailAddress' failed:

Object reference not set to an instance of an object.

System.NullReferenceException
   at ContactList.Tests.TestingConvention.ExecuteAsync(TestClass testClass) in H:\open-source\fixie.demo\src\ContactList.Tests\TestingConvention.cs:line 26
   at Fixie.TestAssembly.RunAsync() in /home/runner/work/fixie/fixie/src/Fixie/TestAssembly.cs:line 60

Test 'ContactList.Tests.Features.Contact.EditContactTests.ShouldRequireValidEmailAddress' skipped:
This test did not run.

Test 'ContactList.Tests.Features.Contact.EditContactTests.ShouldSaveChangesToContactData' failed:

Object reference not set to an instance of an object.

System.NullReferenceException
   at ContactList.Tests.TestingConvention.ExecuteAsync(TestClass testClass) in H:\open-source\fixie.demo\src\ContactList.Tests\TestingConvention.cs:line 26
   at Fixie.TestAssembly.RunAsync() in /home/runner/work/fixie/fixie/src/Fixie/TestAssembly.cs:line 60

Test 'ContactList.Tests.Features.Contact.EditContactTests.ShouldSaveChangesToContactData' skipped:
This test did not run.

7 passed, 12 failed, 5 skipped, took 2.44 seconds


Error executing command: dotnet fixie --configuration $configuration --no-build

Build Failed
```

chan18 commented:
is there a way to debug the test cases(ContactList.Tests) @plioi in visual studio?

plioi commented:
@chan18 This comment is less about this PR and more about the general work to use the beta.

I have produced a new beta, BUT it is proving too problematic. That's my problem, not yours. So, I think if you kept trying at this right now while I'm fixing up the beta, that would be a waste of your time.

Your efforts to upgrade these repos have given me lots of important feedback, so I'm grateful for that, but I think now is a good time to hit the "pause" button while I regroup. "3.0.0 beta 4" will be worth trying again and I'll let you know how that goes.
